enhancements/oc/cli_manager.md

```diff
@@ -0,0 +1,137 @@
+---
+title: cli-manager
+authors:
+  - "@sallyom"
+reviewers:
+  - TBD
+  - "@soltysh"
+approvers:
+  - TBD
+  - "@soltysh"
+creation-date: 2019-12-03
+last-updated: 2019-12-03
+status: provisional|implementable|implemented|deferred|rejected|withdrawn|replaced
+---
+
+# cli-manager
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Open Questions [optional]
+
+Will CLI manager be supported in disconnected environments?  Currently, we provide disconnected environments 
+cli-artifacts and oc download links that do not require anything outside their cluster. What tools/plugins
+need to be managed in disconnected? 
+
+What CLIs will be managed?    
+Where will binaries be published?    
+[Signing binaries for macOS Catalina](https://developer.apple.com/news/?id=09032019a)    
+[related to above, Go toolchain issue with macOS Catalina](https://github.com/golang/go/issues/34986)    
+
+## Summary
+
+This document is a proposal for a component to manage various tools available for OpenShift and related services.
+A CLI manager is desired so that a user can discover, install and upgrade tools easily and from a single
+location.  Each component is responsible for serving their files and registering information regarding supplied binaries
+to a central official location.  The CLI manager will reach out to the central place and retrieve all the binaries from there.
+Similarly, [kubectl plugin manager](https://github.com/GoogleContainerTools/krew/) has an [index](https://github.com/kubernetes-sigs/krew-index)
+to maintain all known krew plugins. 
+
+## Motivation
+
+As more services are created on top of OpenShift, more CLIs are introduced to simplify interaction with these services.
+Some current examples are oc, kubectl, odo, istio, tekton, and knative.  It is difficult for users to discover what tools exist, 
+where to download them from and which version they should download.  We need to simplify as much as possible the interaction
+of services on OpenShift.  This is the motivation for the CLI manager.
```

Comment:
Additionally, more consumers (see my previous wording on it) might come into play and will be able to re-use this mechanism.

Reply:
I've added to 'Motivation'